src/models/ReportModel.ts

```diff
@@ -0,0 +1,52 @@
+import {Column, CreateDateColumn, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
+import { Uuid } from "../types";
+import { UserModel } from "./UserModel";
+import { PostModel } from "./PostModel";
+import { MessageModel } from "./MessageModel";
+import { Report } from "../types";
+
+@Entity("Report")
+export class ReportModel {
```

Comment:
We could add a constructor to this class so when we make an object we don't manually have to set all the attributes (would ensure we don't forget to initialize any attribute and also improve encapsulation a bit). We should then eventually add constructors to the other models in the codebase as well.